src/tools/tidy/src/no_merge.rs

```diff
@@ -0,0 +1,73 @@
+use std::process::Command;
+
+pub fn check(_: (), bad: &mut bool) {
+    let Ok(remote) = get_rust_lang_rust_remote() else { return; };
+    let Ok(merge_commits) = find_merge_commits(&remote) else { return; };
+
+    let mut bad_merge_commits = merge_commits
+        .lines()
+        .filter(|commit| {
+            !(
+                // Bors is the ruler of merge commits.
+                commit.starts_with("Auto merge of")
+                || commit.starts_with("Rollup merge of")
+                // Our fellow subtrees are fine as well.
+                || commit.contains("Merge from rustc")
+                || commit.contains("merge rustc history")
+                || commit.contains("rust-analyzer"))
+                || commit.contains("clippy")
```

Comment:
are these  brackets correct? are these the only tools that cause problems? should there be a note to add future tools to this?

Reply:
The brackets are not correct, thanks, blame rustfmt for not putting the closing one on a new line :D

And yes there should probably be a note for that